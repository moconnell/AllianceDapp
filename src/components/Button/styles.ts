import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background: #ff3864;
  border-radius: 6px;
  cursor: pointer;

  :hover {
    opacity: 0.7;
  }

  ${({ disabled }) =>
    disabled &&
    `
      cursor: auto;
      opacity: 0.7;

      :hover{
      opacity: 0.7;
      }
    `}
`;

export const ButtonComponent = styled.button`
  padding: 8px 16px;
  color: #fff;
  font-family: Inter;
  font-size: 14px;
  font-weight: 700;
  line-height: 20px;
  cursor: inherit;
`;
