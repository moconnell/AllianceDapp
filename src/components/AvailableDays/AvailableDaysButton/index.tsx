import "./styles.css";
import React, { useState } from "react";
import DayOfWeek from "../../../types/dayOfWeek";

interface AvailableDaysButtonProps {
  day: DayOfWeek;
  selected?: boolean;
  onSelect?: (day: DayOfWeek, selected: boolean) => void;
}

const AvailableDaysButton: React.VFC<AvailableDaysButtonProps> = ({
  day,
  selected = false,
  onSelect: selectedChanged,
}) => {
  const [active, setActive] = useState(selected);

  const handleClick = () => {
    const newActive = !active;
    setActive(newActive);
    if (selectedChanged) selectedChanged(day, newActive);
  };

  return (
    <div className="circle-wrapper" onClick={handleClick}>
      <div className={active ? "name-wrapper-clicked" : "name-wrapper"}>
        <span className={active ? "clicked-dot" : "dot"}></span>
        {DayOfWeek[day].substring(0, 3)}
      </div>
    </div>
  );
};

export default AvailableDaysButton;
