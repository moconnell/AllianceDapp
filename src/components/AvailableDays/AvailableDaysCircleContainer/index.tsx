import "./styles.css";
import React, { useState } from "react";

const AvailableDaysButton: React.VFC<{ name: string }> = ({ name }) => {
  const [active, setActive] = useState(false);

  const handleClick = () => {
    setActive(!active);
  };

  return (
    <div className="circle-wrapper" onClick={handleClick}>
      <div className={active ? "name-wrapper-clicked" : "name-wrapper"}>
        <span className={active ? "clicked-dot" : "dot"}></span>
        {name}
      </div>
    </div>
  );
};

export default AvailableDaysButton;
