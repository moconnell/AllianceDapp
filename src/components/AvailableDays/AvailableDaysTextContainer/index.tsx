import "./styles.css";
import React from "react";

export default class AvailableDaysText extends React.Component {
  render() {
    return (
      // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
      <div className="available-days-wrapper">
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        <div className="available-days-text">Available Days</div>
      </div>
    );
  }
}
