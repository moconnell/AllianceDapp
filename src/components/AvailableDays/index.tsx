import "./styles.css";
import React from "react";

export default class AvailableDaysButton extends React.Component {
  constructor(props) {
    super(props);
    this.clickInput = React.createRef();
    this.handleClick = this.handleClick.bind(this);
    this.state = {
      active: false,
    };
  }

  handleClick = () => {
    const currentState = this.state.active;
    this.setState({ active: !currentState });
  };

  render() {
    return (
      <div className="circle-wrapper">
        <div
          ref={this.clickInput}
          className={
            this.state.active ? "name-wrapper-clicked" : "name-wrapper"
          }
        >
          <span
            ref={this.clickInput}
            onClick={this.handleClick}
            className={this.state.active ? "clicked-dot" : "dot"}
          ></span>
          {this.props.name}
        </div>
      </div>
    );
  }
}
