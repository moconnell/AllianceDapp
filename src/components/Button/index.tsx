import React from "react";
import { Container, ButtonComponent } from "./styles";

interface ButtonProps {
  children?: React.ReactNode;
  title?: string;
  disabled?: boolean;
  type?: "button" | "submit" | "reset";
  onClick?: React.MouseEventHandler;
}

const Button = ({
  children,
  disabled,
  title,
  type = "button",
  onClick,
}: ButtonProps) => {
  return (
    <Container aria-disabled={disabled}>
      <ButtonComponent disabled={disabled} title={title} onClick={onClick} type={type}>
        {children}
      </ButtonComponent>
    </Container>
  );
};

export default Button;
