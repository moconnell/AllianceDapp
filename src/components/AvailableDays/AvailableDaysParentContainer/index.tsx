import React from "react";
import AvailableDaysText from "../AvailableDaysTextContainer";
import AvailableDaysButton from "../AvailableDaysCircleContainer";

const AvailableDaysComponent: React.VFC = () => {
  return (
    <div style={{ display: "flex", flexDirection: "row" }}>
      <AvailableDaysText></AvailableDaysText>
      <AvailableDaysButton name="Monday"></AvailableDaysButton>
      <AvailableDaysButton name="Tuesday"></AvailableDaysButton>
      <AvailableDaysButton name="Wednesday"></AvailableDaysButton>
      <AvailableDaysButton name="Thursday"></AvailableDaysButton>
      <AvailableDaysButton name="Friday"></AvailableDaysButton>
      <AvailableDaysButton name="Saturday"></AvailableDaysButton>
      <AvailableDaysButton name="Sunday"></AvailableDaysButton>
    </div>
  );
};

export default AvailableDaysComponent;
