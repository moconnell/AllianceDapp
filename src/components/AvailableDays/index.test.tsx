import { fireEvent, render } from "@testing-library/react";
import { useState } from "react";
import DaysOfWeek from "../../types/daysOfWeek";
import Button from "../Button";
import AvailableDays from "./index";

const TestComponent = () => {
  const [value, setValue] = useState(DaysOfWeek.MonFri);

  return (
    <>
      <AvailableDays value={value} />
      <Button onClick={() => setValue(DaysOfWeek.Monday)} />
    </>
  );
};

describe("<AvailableDays />", () => {
  it("should render correctly", async () => {
    const { asFragment } = render(<AvailableDays />);
    expect(asFragment()).toMatchSnapshot();
  });

  const testData = [
    {
      defaultValue: DaysOfWeek.Monday,
      clickText: /Tue/i,
      expectedValue: DaysOfWeek.Monday | DaysOfWeek.Tuesday,
    },
    {
      defaultValue: DaysOfWeek.MonFri,
      clickText: /Fri/i,
      expectedValue:
        DaysOfWeek.Monday |
        DaysOfWeek.Tuesday |
        DaysOfWeek.Wednesday |
        DaysOfWeek.Thursday,
    },
  ];

  testData.forEach(({ defaultValue, clickText, expectedValue }) =>
    it("should send value onChange", async () => {
      const mockHandleChange = jest.fn();
      const { getByText } = render(
        <AvailableDays value={defaultValue} onChange={mockHandleChange} />
      );
      fireEvent.click(getByText(clickText));
      expect(mockHandleChange).toHaveBeenCalledWith(expectedValue);
    })
  );

  it("should re-render after update", async () => {
    const { asFragment, getByRole, getByText } = render(<TestComponent />);
    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(getByText("Fri"));
    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(getByRole("button"));
    expect(asFragment()).toMatchSnapshot();
  });
});
