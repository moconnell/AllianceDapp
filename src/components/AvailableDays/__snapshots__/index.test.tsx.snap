// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvailableDays /> should re-render after update 1`] = `
<DocumentFragment>
  <div
    class="css-motgx2"
  >
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper-clicked"
      >
        <span
          class="clicked-dot"
        />
        Mon
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper-clicked"
      >
        <span
          class="clicked-dot"
        />
        Tue
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper-clicked"
      >
        <span
          class="clicked-dot"
        />
        Wed
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper-clicked"
      >
        <span
          class="clicked-dot"
        />
        Thu
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper-clicked"
      >
        <span
          class="clicked-dot"
        />
        Fri
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Sat
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Sun
      </div>
    </div>
  </div>
  <div
    class="sc-bczRLJ bbkjqh"
  >
    <button
      class="sc-gsnTZi kKvmkM"
      type="button"
    />
  </div>
</DocumentFragment>
`;

exports[`<AvailableDays /> should re-render after update 2`] = `
<DocumentFragment>
  <div
    class="css-motgx2"
  >
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper-clicked"
      >
        <span
          class="clicked-dot"
        />
        Mon
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper-clicked"
      >
        <span
          class="clicked-dot"
        />
        Tue
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper-clicked"
      >
        <span
          class="clicked-dot"
        />
        Wed
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper-clicked"
      >
        <span
          class="clicked-dot"
        />
        Thu
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Fri
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Sat
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Sun
      </div>
    </div>
  </div>
  <div
    class="sc-bczRLJ bbkjqh"
  >
    <button
      class="sc-gsnTZi kKvmkM"
      type="button"
    />
  </div>
</DocumentFragment>
`;

exports[`<AvailableDays /> should re-render after update 3`] = `
<DocumentFragment>
  <div
    class="css-motgx2"
  >
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper-clicked"
      >
        <span
          class="clicked-dot"
        />
        Mon
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Tue
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Wed
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Thu
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Fri
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Sat
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Sun
      </div>
    </div>
  </div>
  <div
    class="sc-bczRLJ bbkjqh"
  >
    <button
      class="sc-gsnTZi kKvmkM"
      type="button"
    />
  </div>
</DocumentFragment>
`;

exports[`<AvailableDays /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-motgx2"
  >
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Mon
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Tue
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Wed
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Thu
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Fri
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Sat
      </div>
    </div>
    <div
      class="circle-wrapper"
    >
      <div
        class="name-wrapper"
      >
        <span
          class="dot"
        />
        Sun
      </div>
    </div>
  </div>
</DocumentFragment>
`;
